core/services/keeper/upkeep_executer.go

```diff
@@ -194,7 +194,8 @@ func (ex *UpkeepExecuter) execute(upkeep UpkeepRegistration, head *evmtypes.Head
 
 		// Make sure the gas price is at least as large as the basefee to avoid ErrFeeCapTooLow error from geth during eth call.
 		// If head.BaseFeePerGas, we assume it is a EIP-1559 chain.
-		if head.BaseFeePerGas != nil && head.BaseFeePerGas.ToInt().BitLen() > 0 {
+		// Note: gasPrice will be null if EvmEIP1559DynamicFees is enabled.
+		if gasPrice != nil && head.BaseFeePerGas != nil && head.BaseFeePerGas.ToInt().BitLen() > 0 {
```

Comment:
Should we be setting gasPrice = baseFee in case it's nil? (by skipping the compare)

Reply:
nvm, I think I understand the logic that either gasPrice is required or gasTipCap + gasFeeCap

Reply:
nvm, I think I understand the logic that either gasPrice is required or gasTipCap + gasFeeCap